cmd/internal/cli/actions.go

```diff
@@ -140,10 +140,17 @@ func handleLibrary(u, libraryURL string) (string, error) {
 
 	imageRef := libraryhelper.NormalizeLibraryRef(u)
 
-	libraryImage, _, err := c.GetImage(ctx, imageRef)
+	libraryImage, existOk, err := c.GetImage(ctx, imageRef)
 	if err != nil {
 		return "", err
 	}
+	if !existOk {
```

Comment:
this is fine, but why is a non-existent / failure to get an image giving err == nil?

Skimming thru the code either err == nil and found == true, in which case libraryImage will have _something_, or err == nil and found == false, and libraryImage will be nil.

_I would prefer_ to not have the extra `if libraryImage == nil` check, as it makes reading the code a little harder, and as the comment states, that path is never hit.

It's interesting that the code goes thru several layers of handling "not exist" as an error or not. At some level, not found is in fact a non-nil error, which gets translated to a nil error one layer above it. It would be more idiomatic to report not found as an error.

Reply:
Okay, I'll remove the `if libraryImage == nil {...}` check :+1: